File: Assignment3/SalesInterior/SalesInterior/Models/House.swift
import Foundation
import FirebaseFirestore

/// House document - stored in "houses" collection
/// Rooms are stored in a subcollection: houses/{houseId}/rooms
struct House: Identifiable, Codable {
    @DocumentID var id: String?
    var clientName: String
    var projectCode: String
    var street: String
    var city: String
    var postcode: String

    enum CodingKeys: String, CodingKey {
        case id, clientName, projectCode, street, city, postcode
    }

    init(
        clientName: String = "",
        projectCode: String = "",
        street: String = "",
        city: String = "",
        postcode: String = ""
    ) {
        self.clientName = clientName
        self.projectCode = projectCode
        self.street = street
        self.city = city
        self.postcode = postcode
    }
    
    // Remove custom decoder - let @DocumentID handle it automatically
}

File: Assignement4/sales_interior/lib/models/house.dart
class House {
  final String id;
  final String clientName;
  final String projectCode;
  final String street;
  final String city;
  final String postcode;

  House({
    required this.id,
    required this.clientName,
    required this.projectCode,
    required this.street,
    required this.city,
    required this.postcode,
  });

  factory House.fromMap(Map<String, dynamic> map, String id) {
    return House(
      id: id,
      clientName: map['clientName'] as String? ?? '',
      projectCode: map['projectCode'] as String? ?? '',
      street: map['street'] as String? ?? '',
      city: map['city'] as String? ?? '',
      postcode: map['postcode'] as String? ?? '',
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'clientName': clientName,
      'projectCode': projectCode,
      'street': street,
      'city': city,
      'postcode': postcode,
    };
  }

  @override
  String toString() {
    return 'House(id: $id, clientName: $clientName, projectCode: $projectCode, street: $street, city: $city, postcode: $postcode)';
  }
}
